packages/controllers/src/snaps/utils.ts

```diff
@@ -356,8 +360,14 @@ async function fetchNpmTarball(
     );
   }
 
+  // Override the tarball hostname/protocol with registryUrl hostname/protocol
+  const newRegistryUrl = new URL(registryUrl);
+  const newTarballUrl = new URL(tarballUrlString);
+  newTarballUrl.hostname = newRegistryUrl.hostname;
+  newTarballUrl.protocol = newRegistryUrl.protocol;
```

Comment:
Nit: This would be simpler, and would add support for custom registries on non-standard ports:

```suggestion
  newTarballUrl.origin = newRegistryUrl.origin;
```

Reply:
origin is read-only 😞 https://developer.mozilla.org/en-US/docs/Web/API/URL/origin